source/Parser.cpp

```diff
@@ -1907,6 +1914,125 @@ ParameterDeclarationSyntax* Parser::parseParameterPort() {
     return alloc.emplace<ParameterDeclarationSyntax>(Token(), type, parseOneVariableDeclarator());
 }
 
+ClockingSkewSyntax* Parser::parseClockingSkew() {
+    Token edge, hash;
+    ExpressionSyntax* value = nullptr;
+    if (peek().kind == TokenKind::EdgeKeyword || peek().kind == TokenKind::PosEdgeKeyword || peek().kind == TokenKind::NegEdgeKeyword) {
+        edge = consume();
+    }
+    if (peek().kind == TokenKind::Hash) {
+        hash = consume();
+        switch(peek().kind) {
+            case TokenKind::OpenParenthesis:
+            {
+                auto openParen = consume();
+                auto innerExpr = parseMinTypMaxExpression();
+                auto closeParen = expect(TokenKind::CloseParenthesis);
+                value = alloc.emplace<ParenthesizedExpressionSyntax>(openParen, innerExpr, closeParen);
+                break;
+            }
+            case TokenKind::IntegerLiteral:
+            case TokenKind::RealLiteral:
+            case TokenKind::TimeLiteral:
+            case TokenKind::OneStep:
+            {
+                auto literal = consume();
+                value = alloc.emplace<LiteralExpressionSyntax>(getLiteralExpression(literal.kind), literal);
+                break;
+            }
+            default:
+                value = alloc.emplace<IdentifierNameSyntax>(expect(TokenKind::Identifier));
+                break;
+        }
+    }
+    if (!edge && !hash) {
+        addError(DiagCode::ExpectedClockingSkew, peek().location());
+    }
+    return alloc.emplace<ClockingSkewSyntax>(edge, hash, value);
+}
+
+ClockingDeclarationSyntax* Parser::parseClockingDeclaration(ArrayRef<AttributeInstanceSyntax*> attributes) {
+    Token globalOrDefault;
+    if (peek().kind != TokenKind::ClockingKeyword) {
+        globalOrDefault = consume();
+    }
+    Token clocking = expect(TokenKind::ClockingKeyword);
+    Token blockName = expect(TokenKind::Identifier);
+    Token at, eventIdentifier;
+    ParenthesizedEventExpressionSyntax* event = nullptr;
+    if (peek().kind == TokenKind::At) {
+        at = consume();
+        if (peek().kind == TokenKind::OpenParenthesis) {
+            auto openParen = consume();
+            auto innerExpr = parseEventExpression();
+            auto closeParen = expect(TokenKind::CloseParenthesis);
+            event = alloc.emplace<ParenthesizedEventExpressionSyntax>(openParen, innerExpr, closeParen);
+        } else {
+            eventIdentifier = expect(TokenKind::Identifier);
+        }
+    }
+    Token semi = expect(TokenKind::Semicolon);
+    SmallVectorSized<ClockingItemSyntax*, 4> buffer;
+    if (globalOrDefault.kind != TokenKind::GlobalKeyword) {
+        while(peek().kind != TokenKind::EndClockingKeyword && peek().kind != TokenKind::EndOfFile) {
```

Comment:
parseBlockItems might be a better example in that it also checks isEndKeyword(), which means if the user forgot the endclocking but has some other end token in their file (like endmodule or whatever) we won't just suck up every token.